### PR TITLE
msg: remove xiomessenger

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
> looks fine to me, I think there should be flexibility for multiple messengers, however

Totally agree.  I just don't want to have to update essentially dead code everything time we make an adjustment to the interface.

File: src/test/objectstore/CMakeLists.txt [hunk 1]
```diff
@@ -124,7 +124,6 @@ endif(WITH_BLUESTORE)
 add_executable(unittest_transaction
   test_transaction.cc)
 add_ceph_unittest(unittest_transaction)
-target_link_libraries(unittest_transaction os ceph-common ${XIO_LIBRARY})
```
Comment (tchaikov) [inline src/test/objectstore/CMakeLists.txt]: we still need to link against `os` and `ceph-common`.

**Comment by mattbenjamin** [comment]
I didnt see where DirectMessenger was removed -- I would like to preserve and use that, folks

**Comment by gregsfortytwo** [comment]
I don't see anything in a quick skim that makes it hard to keep the direct messenger; I think @liewegas just wiped out the whole set of src/test/messenger directories.

**Comment by mattbenjamin** [comment]
@tchaikov I repulled your version, found an un-fixed inclusion of MDataPing.h, updated again;  builds for me (not building MDS or RBD)

**Comment by tchaikov** [comment]
pushed to ceph/ceph-ci:wip-27021 to verify it the packages build.

see https://shaman.ceph.com/builds/ceph/wip-27021/55867eccd40fdf79ec3acfaa596f489c1077769b/

File: src/test/objectstore/CMakeLists.txt (hunk 1 above)
Comment (markhpc) [reply src/test/objectstore/CMakeLists.txt]: I'm in favor of dropping xiomessenger, and I'm fairly ambivalent about direct_messenger as I haven't been using it.  If it's useful to folks (and potentially is useful to me) I'd be in favor of keeping it if it's not a lot of overhead to keep it around.

File: src/test/messenger/CMakeLists.txt [hunk 2]
```diff
@@ -1,43 +0,0 @@
-add_executable(simple_server
```
Comment (tchaikov) [inline src/test/messenger/CMakeLists.txt]: we need to remove `add_subdirectory(messenger)` in `src/test/CMakeLists.txt`, just pushed the updated change to this PR.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-03-20_05:45:48-rados-wip-kefu-testing-2019-03-20-1120-distro-basic-smithi/

```
    ceph:
      conf:
        client:
          ms type: simple
...
2019-03-20T08:26:16.431 INFO:teuthology.orchestra.run.smithi085:> sudo ceph --cluster ceph osd crush tunables hammer
2019-03-20T08:26:16.490 INFO:tasks.ceph.mon.c.smithi139.stderr:warning: line 40: 'osd_pool_default_size' in section 'global' redefined
2019-03-20T08:26:16.505 INFO:tasks.ceph.mgr.y.smithi085.stderr:warning: line 40: 'osd_pool_default_size' in section 'global' redefined
2019-03-20T08:26:16.508 INFO:tasks.ceph.mgr.x.smithi081.stderr:warning: line 40: 'osd_pool_default_size' in section 'global' redefined
2019-03-20T08:26:16.565 INFO:teuthology.orchestra.run.smithi085.stderr:/build/ceph-14.2.0-165-gba7267b/src/mon/MonClient.cc: In function 'int MonClient::get_monmap_and_config()' thread 7f0c7f4fe700 time 2019-03-20 08:26:16.569069
2019-03-20T08:26:16.565 INFO:teuthology.orchestra.run.smithi085.stderr:/build/ceph-14.2.0-165-gba7267b/src/mon/MonClient.cc: 124: FAILED ceph_assert(messenger)
2019-03-20T08:26:16.565 INFO:teuthology.orchestra.run.smithi085.stderr:2019-03-20 08:26:16.565 7f0c7f4fe700 -1 unrecognized ms_type 'simple'
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: ceph version 14.2.0-165-gba7267b (ba7267bddd2924864acb73f05dd62c7dd7cfce76) nautilus (stable)
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x152) [0x7f0c82c45a12]
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 2: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x7f0c82c45bed]
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 3: (MonClient::get_monmap_and_config()+0xac9) [0x7f0c82fa1d59]
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 4: (()+0x96e3f) [0x7f0c8bf0ce3f]
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 5: (rados_connect()+0x1d) [0x7f0c8bea8d3d]
2019-03-20T08:26:16.566 INFO:teuthology.orchestra.run.smithi085.stderr: 6: (()+0x307b9) [0x7f0c8c1f47b9]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 7: (PyEval_EvalFrameEx()+0x729e) [0x4c30ce]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 8: (PyEval_EvalCodeEx()+0x306) [0x4b9ab6]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 9: (PyEval_EvalFrameEx()+0x603f) [0x4c1e6f]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 10: (PyEval_EvalFrameEx()+0x553f) [0x4c136f]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 11: (PyEval_EvalCodeEx()+0x306) [0x4b9ab6]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 12: /usr/bin/python2.7() [0x4d54b9]
2019-03-20T08:26:16.567 INFO:teuthology.orchestra.run.smithi085.stderr: 13: /usr/bin/python2.7() [0x4eebee]
2019-03-20T08:26:16.568 INFO:teuthology.orchestra.run.smithi085.stderr: 14: (PyObject_Call()+0x3e) [0x4a577e]
2019-03-20T08:26:16.568 INFO:teuthology.orchestra.run.smithi085.stderr: 15: (PyEval_CallObjectWithKeywords()+0x30) [0x4c5e10]
2019-03-20T08:26:16.568 INFO:teuthology.orchestra.run.smithi085.stderr: 16: /usr/bin/python2.7() [0x589172]
2019-03-20T08:26:16.568 INFO:teuthology.orchestra.run.smithi085.stderr: 17: (()+0x76ba) [0x7f0c8d87c6ba]
2019-03-20T08:26:16.568 INFO:teuthology.orchestra.run.smithi085.stderr: 18: (clone()+0x6d) [0x7f0c8d5b241d]
````

can we only override the "client.2" 's "ms_type"?

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-03-20_05:45:48-rados-wip-kefu-testing-2019-03-20-1120-distro-basic-smithi/3751812/

```
2019-03-20T08:25:21.900 INFO:tasks.workunit.client.0.smithi046.stderr:+ ceph config set mon.a debug_xio 22
2019-03-20T08:25:22.156 INFO:tasks.workunit.client.0.smithi046.stderr:Error EINVAL: unrecognized config option 'debug_xio'
```
also we need to remove all occurrences of "debug_xio".

**Comment by liewegas** [comment]
Instead I'll just drop that line so that we end up with the default

**Comment by liewegas** [comment]
hrm, no, that doesn't work either.

i think we can either

- drop hammer
- drop the msgr variation and stick with default.  in practice that means we don't test both v2+v1 and v1 only.

what do you think?

**Comment by tchaikov** [comment]
> drop the msgr variation and stick with default.  in practice that means we don't test both v2+v1 and v1 only.

do we have any test exercising the combination of v2+v1 and v1 after this change of "use default messenger" ? i think it's a typical setting in quite a few clusters where they need to support old clients and also have the luxury of a recent ceph installation.

**Comment by gregsfortytwo** [comment]
Yes, it will be important to continue testing v1 in combinations with v2. And hopefully older implementations of it to make sure we didn't break something...
Why does dropping hammer help? Though doing so shouldn't be an issue at this point, unless I'm missing something.

**Comment by tchaikov** [comment]
> Why does dropping hammer help?

i guess the reason is that we removed the experimental tag of async messenger in jewel (?) i guess? so we don't need to override the "ms_type" just for backward compatibility with hammer.

**Comment by liewegas** [comment]
the latest version of the patch fixes these issues by collapsing the client version and msgr choice into a single facet.

also adds mimic and nautilus.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-03-21_02:00:24-rados-wip-kefu-testing-2019-03-21-0010-distro-basic-mira/

File: src/msg/Message.h [hunk 3]
```diff
@@ -187,8 +187,7 @@
 #define MSG_CRC_ALL            (MSG_CRC_DATA | MSG_CRC_HEADER)
 
 // Xio Testing
-#define MSG_DATA_PING		  0x602
-
+//#define MSG_DATA_PING		  0x602
```
Comment (tchaikov) [inline src/msg/Message.h]: we can recycle this message id after two releases, right? but, yeah, it's not a big deal .

**Comment by gregsfortytwo** [comment]
I don’t think we can genetically recycle message IDs as we have to support
old clients correctly.

**Pull request merged**